### PR TITLE
Removing out-of-date boards

### PR DESCRIPTION
**Pull request by andrey4latyshev**
# Description

Edison and Gallileo boards will be no longer supported in IoT Samples.

Fixes https://jira.devtools.intel.com/browse/DEV1A-1542 

## Type of change

Please delete options that are not relevant. Add a 'X' to the one that is applicable. 

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Sample Migration (Moving sample from old repository after completing checklist established)

# Checklist for Moving samples:
Links and Details can be found in the samples WG Teams Files. 

- [ ] Review sample design with domain reviewers https://github.com/oneapi-src/oneAPI-samples/wiki/Reviewers-and-Domain-Experts 
- [ ] Implement coding guidelines and ensure code quality.
- [ ] Adhere to sample.json specification. https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-specification
- [ ] Run jsonlint on sample.json to verify json syntax. www.jsonlint.com
- [ ] Adhere to readme template 
- [ ] Ensure/create CI test configurations for sample (ciTests field) https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-ci-test-object
- [ ] Enforce format via clang-format config file
- [ ] Review DPC++ code with Paul Peterseon. (GitHub User: pmpeter1)
- [ ] Review readme with Tom Lenth or Joe Oster. (GitHub User: JoeOster)
- [ ] Tested using Dev Cloud when applicable
- [ ] Implement fixes for ONSAM Jiras
- [ ] If you have new dependencies/binaries, inform Samples Project Manager Swapna R Dontharaju (@srdontha) or @JoeOster

### CODE REVIEW

File: DirectProgramming/DPC++FPGA/ReferenceDesigns/gzip/Zlib_license.txt [hunk 1]
```diff
@@ -1,3 +1,5 @@
+zlib License
```
Comment (anjgola) [inline DirectProgramming/DPC++FPGA/ReferenceDesigns/gzip/Zlib_license.txt:1]: Why this change

**Review by anjgola — DISMISSED** [review]
I was not sure about the rationale for this change because the case break statement are more for protection from running the sample incorrectly if you end up running the sample on these boards so it is not necessarily ‘supporting these boards’ but warning a user if they run the sample on these boards.

Since engineering team is recommending this, approving.

**Review by JoeOster — APPROVED** [review]
